src/adapters/aol.js

```diff
@@ -1,183 +1,141 @@
-   * @param {String} id to identify the div
-   * @return {String} the id used with the div
-   */
-  function _dummyUnit(id) {
-    var div = d.createElement('DIV');
-
-    if (!id || !id.length) {
-      id = 'ad-placeholder-' + (++dummyUnitIdCount);
-    }
-
-    div.id = id + '-head-unit';
-    h.appendChild(div);
-    return div.id;
+  function template(strings, ...keys) {
+    return function(...values) {
+      let dict = values[values.length - 1] || {};
+      let result = [strings[0]];
+      keys.forEach(function(key, i) {
+        let value = Number.isInteger(key) ? values[key] : dict[key];
+        result.push(value, strings[i + 1]);
+      });
+      return result.join('');
+    };
   }
 
-  /**
-   * @private Add a succesful bid response for aol
-   * @param {ADTECHResponse} response the response for the bid
-   * @param {ADTECHContext} context the context passed from aol
-   */
-  function _addBid(response, context) {
-    var bid = bidsMap[context.alias];
-    var cpm;
+  function _buildPubapiUrl(bid) {
+    const params = bid.params;
+    const serverParam = params.server;
+    var regionParam = params.region || 'us';
+    var server;
 
-    if (!bid) {
-      utils.logError('mismatched bid: ' + context.placement, ADTECH_BIDDER_NAME, context);
-      return;
+    if (!SERVER_MAP.hasOwnProperty(regionParam)) {
+      console.warn(`Unknown region '${regionParam}' for AOL bidder.`);
+      regionParam = 'us'; // Default region.
     }
 
-    cpm = response.getCPM();
-    if (cpm === null || isNaN(cpm)) {
-      return _addErrorBid(response, context);
+    if (serverParam) {
+      server = serverParam;
+    } else {
+      server = SERVER_MAP[regionParam];
     }
 
-    // clean up--we no longer need to store the bid
-    delete bidsMap[context.alias];
-
-    var bidResponse = bidfactory.createBid(1);
-    var ad = response.getCreative();
-    if (typeof response.getPixels() !== 'undefined') {
-      ad += response.getPixels();
-    }
-    bidResponse.bidderCode = ADTECH_BIDDER_NAME;
-    bidResponse.ad = ad;
-    bidResponse.cpm = cpm;
-    bidResponse.width = response.getAdWidth();
-    bidResponse.height = response.getAdHeight();
-    bidResponse.creativeId = response.getCreativeId();
+    // Set region param, used by AOL analytics.
+    params.region = regionParam;
+
+    return pubapiTemplate({
+      protocol: (document.location.protocol === 'https:') ? 'https' : 'http',
+      host: server,
+      network: params.network,
+      placement: parseInt(params.placement),
+      pageid: params.pageId || 0,
+      sizeid: params.sizeId || 0,
+      alias: params.alias || utils.getUniqueIdentifierStr(),
+      bidfloor: (typeof params.bidFloor !== 'undefined') ?
+        `;bidfloor=${params.bidFloor.toString()}` : '',
+      misc: new Date().getTime() // cache busting
+    });
+  }
 
-    // add it to the bid manager
+  function _addErrorBidResponse(bid, response = {}) {
+    var bidResponse = bidfactory.createBid(2);
```

Comment:
Consider setting the bid response ID to match the bid request ID as described here: #509.

Doing so will position the adapter to take advantage of Prebid Roadmap features such as secure iframe support, improved analytics, Prebid Auctions and so on.